gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/GigawordParserUtils.java

```diff
@@ -0,0 +1,126 @@
+package edu.isi.vista.gigawordIndexer;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.commons.io.FilenameUtils;
+
+public class GigawordParserUtils {
+
+  // start of a new document
+  private static Pattern GIGAWORD_DOC_ELEMENT_PATTERN = Pattern.compile("<DOC id=\"(.*?)\".*>");
+
+  // end of a document
+  private static final String END_OF_DOCUMENT_MARKER = "</DOC>";
+
+  public GigawordParserUtils() {}
+
+  /**
+   * Parse the full text from a GZip file
+   *
+   * @param filePath
+   * @return String full text
+   * @throws FileNotFoundException
+   * @throws IOException
+   */
+  private static String gZipFileToString(String filePath)
+      throws FileNotFoundException, IOException {
+    StringBuilder fullText = new StringBuilder();
+    BufferedReader reader =
+        new BufferedReader(
+            new InputStreamReader(new GZIPInputStream(new FileInputStream(filePath))));
+
+    // first line
+    String line = reader.readLine();
+    fullText.append(line);
+
+    while ((line = reader.readLine()) != null) {
+      fullText.append("\n" + line);
+    }
+    reader.close();
+    return fullText.toString();
+  }
+
+  /**
+   * split full text into separate articles
+   *
+   * @param fullText
+   * @return A HashMap of document ID and article
+   */
+  private static HashMap<String, String> splitTextIntoArticles(String fullText) {
```

Comment:
I wouldn't change it now, but an alternate approach to consider is just returning an `Iterator` over the articles in the file, since you don't have any need for the random access nature of the `Map`. That would let you avoid loading the entire files into memory.  (In this case we know the files are small enough we can get away with it)

Reply:
I don't understand what you mean in the first comment, can you explain?